diff --git a/mod/quiz/accessrule/ipaddress/tests/rule_test.php b/mod/quiz/accessrule/ipaddress/tests/rule_test.php
index 007000a..3f8f520 100644
--- a/mod/quiz/accessrule/ipaddress/tests/rule_test.php
+++ b/mod/quiz/accessrule/ipaddress/tests/rule_test.php
@@ -56,7 +56,8 @@ class quizaccess_ipaddress_testcase extends basic_testcase {
             $this->assertFalse($rule->description());
             $this->assertFalse($rule->prevent_new_attempt(0, $attempt));
             $this->assertFalse($rule->is_finished(0, $attempt));
-            $this->assertFalse($rule->time_left($attempt, 1));
+            $this->assertFalse($rule->end_time($attempt));
+            $this->assertFalse($rule->time_left_display($attempt, 0));
         }
 
         $quiz->subnet = '0.0.0.0';
@@ -68,6 +69,7 @@ class quizaccess_ipaddress_testcase extends basic_testcase {
         $this->assertEmpty($rule->description());
         $this->assertFalse($rule->prevent_new_attempt(0, $attempt));
         $this->assertFalse($rule->is_finished(0, $attempt));
-        $this->assertFalse($rule->time_left($attempt, 1));
+        $this->assertFalse($rule->end_time($attempt));
+        $this->assertFalse($rule->time_left_display($attempt, 0));
     }
 }
