diff --git a/mnet/service/enrol/course.php b/mnet/service/enrol/course.php
index d67eaf6..632e4d3 100644
--- a/mnet/service/enrol/course.php
+++ b/mnet/service/enrol/course.php
@@ -62,7 +62,7 @@ echo $OUTPUT->heading($icon . s($course->fullname));
 
 // collapsible course summary
 if (!empty($course->summary)) {
-    unset($options);
+    $options = new stdClass();
     $options->trusted = false;
     $options->para    = false;
     $options->filter  = false;
