diff --git a/theme/canvas/layout/general.php b/theme/canvas/layout/general.php
index cda6005..5261314 100644
--- a/theme/canvas/layout/general.php
+++ b/theme/canvas/layout/general.php
@@ -3,17 +3,27 @@
 $hasheading = ($PAGE->heading);
 $hasnavbar = (empty($PAGE->layout_options['nonavbar']) && $PAGE->has_navbar());
 $hasfooter = (empty($PAGE->layout_options['nofooter']));
-$hassidepre = $PAGE->blocks->region_has_content('side-pre', $OUTPUT);
-$hassidepost = $PAGE->blocks->region_has_content('side-post', $OUTPUT);
+$hassidepre = (empty($PAGE->layout_options['noblocks']) && $PAGE->blocks->region_has_content('side-pre', $OUTPUT));
+$hassidepost = (empty($PAGE->layout_options['noblocks']) && $PAGE->blocks->region_has_content('side-post', $OUTPUT));
+$haslogininfo = (empty($PAGE->layout_options['nologininfo']));
+
+$showsidepre = ($hassidepre && !$PAGE->blocks->region_completely_docked('side-pre', $OUTPUT));
+$showsidepost = ($hassidepost && !$PAGE->blocks->region_completely_docked('side-post', $OUTPUT));
+
+$custommenu = $OUTPUT->custom_menu();
+$hascustommenu = (empty($PAGE->layout_options['nocustommenu']) && !empty($custommenu));
 
 $bodyclasses = array();
-if ($hassidepre && !$hassidepost) {
+if ($showsidepre && !$showsidepost) {
     $bodyclasses[] = 'side-pre-only';
-} else if ($hassidepost && !$hassidepre) {
+} else if ($showsidepost && !$showsidepre) {
     $bodyclasses[] = 'side-post-only';
-} else if (!$hassidepost && !$hassidepre) {
+} else if (!$showsidepost && !$showsidepre) {
     $bodyclasses[] = 'content-only';
 }
+if ($hascustommenu) {
+    $bodyclasses[] = 'has_custom_menu';
+}
 
 echo $OUTPUT->doctype() ?>
 <html <?php echo $OUTPUT->htmlattributes() ?>>
@@ -49,6 +59,10 @@ echo $OUTPUT->doctype() ?>
             </div>
         </div>
 
+        <?php if ($hascustommenu) { ?>
+            <div id="custommenu"><?php echo $custommenu; ?></div>
+        <?php } ?>
+
         <?php if ($hasnavbar) { ?>
             <div class="navbar clearfix">
                 <div class="breadcrumb"><?php echo $OUTPUT->navbar(); ?></div>
@@ -76,7 +90,7 @@ echo $OUTPUT->doctype() ?>
                         </div>
 
                         <?php if ($hassidepre) { ?>
-                        <div id="region-pre">
+                        <div id="region-pre" class="block-region">
                             <div class="region-content">
                                 <?php echo $OUTPUT->blocks_for_region('side-pre') ?>
                             </div>
@@ -84,7 +98,7 @@ echo $OUTPUT->doctype() ?>
                         <?php } ?>
 
                         <?php if ($hassidepost) { ?>
-                        <div id="region-post">
+                        <div id="region-post" class="block-region">
                             <div class="region-content">
                                 <?php echo $OUTPUT->blocks_for_region('side-post') ?>
                             </div>
