diff --git a/elis/core/lib/page.class.php b/elis/core/lib/page.class.php
index 7940abf..d316dd8 100644
--- a/elis/core/lib/page.class.php
+++ b/elis/core/lib/page.class.php
@@ -44,7 +44,7 @@ abstract class elis_page extends moodle_page {
     /**
      * Page parameters (if null, use the HTTP parameters)
      */
-    private $params = null;
+    protected $params = null;
 
     /**
      * Constructor.
