diff --git a/elis/core/lib/table.class.php b/elis/core/lib/table.class.php
index 12cd323..d9c7d44 100644
--- a/elis/core/lib/table.class.php
+++ b/elis/core/lib/table.class.php
@@ -182,6 +182,9 @@ class display_table {
 
             $table->data[] = $row;
         }
+        if ($this->items instanceof Iterator) {
+            $this->items->close();
+        }
         $this->table = $table;
     }
 
