diff --git a/grade/export/ods/grade_export_ods.php b/grade/export/ods/grade_export_ods.php
index e366131..48ae321 100644
--- a/grade/export/ods/grade_export_ods.php
+++ b/grade/export/ods/grade_export_ods.php
@@ -41,7 +41,7 @@ class grade_export_ods extends grade_export {
     /// Sending HTTP headers
         $workbook->send($downloadfilename);
     /// Adding the worksheet
-        $myxls =& $workbook->add_worksheet($strgrades);
+        $myxls = $workbook->add_worksheet($strgrades);
 
     /// Print names of all the fields
         $myxls->write_string(0,0,get_string("firstname"));
