diff --git a/version.php b/version.php
index 3126606..411d052 100644
--- a/version.php
+++ b/version.php
@@ -29,11 +29,11 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$version  = 2013051408.00;              // 20130514      = branching date YYYYMMDD - do not modify!
+$version  = 2013051407.00;              // 20130514      = branching date YYYYMMDD - do not modify!
                                         //         RR    = release increments - 00 in DEV branches
                                         //           .XX = incremental changes
 
-$release  = '2.5.8 (Build: 20140908)';  // Human-friendly version name
+$release  = '2.5.7 (Build: 20140714)';  // Human-friendly version name
 
 $branch   = '25';                       // this version's branch
 $maturity = MATURITY_STABLE;            // this version's maturity level
