diff --git a/version.php b/version.php
index 3f48fac..e301f7d 100644
--- a/version.php
+++ b/version.php
@@ -30,10 +30,10 @@
 defined('MOODLE_INTERNAL') || die();
 
 
-$version  = 2011120505.04;              // 20111205      = branching date YYYYMMDD - do not modify!
+$version  = 2011120506.00;              // 20111205      = branching date YYYYMMDD - do not modify!
                                         //         RR    = release increments - 00 in DEV branches
                                         //           .XX = incremental changes
 
-$release  = '2.2.5+ (Build: 20121005)'; // Human-friendly version name
+$release  = '2.2.6 (Build: 20121112)';  // Human-friendly version name
 
 $maturity = MATURITY_STABLE;            // this version's maturity level
