diff --git a/version.php b/version.php
index a5b0b9a..295fdcc 100644
--- a/version.php
+++ b/version.php
@@ -30,11 +30,11 @@
 defined('MOODLE_INTERNAL') || die();
 
 
-$version  = 2012062502.05;              // YYYYMMDD      = weekly release date of this DEV branch
+$version  = 2012062503.00;              // YYYYMMDD      = weekly release date of this DEV branch
                                         //         RR    = release increments - 00 in DEV branches
                                         //           .XX = incremental changes
 
-$release  = '2.3.2+ (Build: 20121005)'; // Human-friendly version name
+$release  = '2.3.3 (Build: 20121112)';  // Human-friendly version name
 
 $branch   = '23';                       // this version's branch
 $maturity = MATURITY_STABLE;            // this version's maturity level
