diff --git a/version.php b/version.php
index 6319a64..fb2a5d2 100644
--- a/version.php
+++ b/version.php
@@ -10,6 +10,6 @@
                               //         X     = release number 1.9.[0,1,2,3,4,5...]
                               //          Y.YY = micro-increments between releases
 
-    $release = '1.9.13+ (Build: 20110811)';     // Human-friendly version name
+    $release = '1.9.13+ (Build: 20110817)';     // Human-friendly version name
 
 ?>
