diff --git a/install/lang/is/admin.php b/install/lang/is/admin.php
index 645f4f0..34600b2 100644
--- a/install/lang/is/admin.php
+++ b/install/lang/is/admin.php
@@ -39,6 +39,5 @@ $string['clitypevaluedefault'] = 'Sláðu inn gildi, sláðu á Enter hnappinn 
 $string['cliunknowoption'] = 'Óþekktir valkostir:
   {$a}
 Vinsamlegast notaðu --help valkostinn.';
-$string['cliyesnoprompt'] = 'skráðu y (þýðir já) eða n (þýðir nei)';
 $string['environmentrequireinstall'] = 'verður að vera uppsett og virkjað';
 $string['environmentrequireversion'] = 'krafist er útgáfu {$a->needed} en þú notast við útgáfu {$a->current}';
