diff --git a/mod/quiz/accessrule/safebrowser/tests/rule_test.php b/mod/quiz/accessrule/safebrowser/tests/rule_test.php
index 6922e99..470ecc9 100644
--- a/mod/quiz/accessrule/safebrowser/tests/rule_test.php
+++ b/mod/quiz/accessrule/safebrowser/tests/rule_test.php
@@ -58,6 +58,7 @@ class quizaccess_safebrowser_testcase extends basic_testcase {
             $rule->description());
         $this->assertFalse($rule->prevent_new_attempt(0, $attempt));
         $this->assertFalse($rule->is_finished(0, $attempt));
-        $this->assertFalse($rule->time_left($attempt, 1));
+        $this->assertFalse($rule->end_time($attempt));
+        $this->assertFalse($rule->time_left_display($attempt, 0));
     }
 }
