diff --git a/elis/core/lib/filtering/dependentselect.php b/elis/core/lib/filtering/dependentselect.php
index 6d2440d..5a3d1b0 100644
--- a/elis/core/lib/filtering/dependentselect.php
+++ b/elis/core/lib/filtering/dependentselect.php
@@ -102,7 +102,7 @@ class generalized_filter_dependentselect extends generalized_filter_type {
         $fullpath = $this->_report_path . $this->_filename;
         $parent   = $this->_uniqueid .'_parent';
 
-        $js = "dependentselect_updateoptions('{$parent}','{$this->_uniqueid}','{$fullpath}');";
+        $js = "dependentselect_updateoptions('{$parent}', '{$this->_uniqueid}', '{$fullpath}');";
 
         $objs = array();
         $objs[] =& $mform->createElement('select', $this->_uniqueid.'_parent', null, $options_array,
@@ -159,7 +159,7 @@ class generalized_filter_dependentselect extends generalized_filter_type {
         }
 
         $param_name = 'ex_dependselect'. $counter++;
-        $value = addslashes($data['value']);
+        $value = $data['value'];
         return array("{$full_fieldname} = :{$param_name}",
                      array($param_name => $value));
     }
