diff --git a/elis/core/lib/filtering/profileselect.php b/elis/core/lib/filtering/profileselect.php
index 5c89d47..18b21d4 100644
--- a/elis/core/lib/filtering/profileselect.php
+++ b/elis/core/lib/filtering/profileselect.php
@@ -185,7 +185,7 @@ class generalized_filter_profileselect extends generalized_filter_type {
         }
 
         $value = $data['value'];
-        $value = addslashes($value);
+        //$value = addslashes($value);
 
         $sql = "{$full_fieldname} IN
                 (SELECT inner_data.userid
