diff --git a/elis/core/lib/filtering/text.php b/elis/core/lib/filtering/text.php
index 58ac6e2..c89ad32 100644
--- a/elis/core/lib/filtering/text.php
+++ b/elis/core/lib/filtering/text.php
@@ -177,7 +177,7 @@ class generalized_filter_text extends generalized_filter_type {
             return null;
         }
 
-        $value = addslashes($data['value']);
+        $value = $data['value'];
 
         switch($data['operator']) {
             case generalized_filter_text::$OPERATOR_CONTAINS:
