diff --git a/mod/assign/submission/comments/locallib.php b/mod/assign/submission/comments/locallib.php
index eb87938..1d91186 100644
--- a/mod/assign/submission/comments/locallib.php
+++ b/mod/assign/submission/comments/locallib.php
@@ -78,12 +78,13 @@ class assign_submission_comments extends assign_submission_plugin {
     }
 
     /**
-     * Always return false because at a minimum there is the comments control
+     * Always return true because the submission comments are not part of the submission form.
+     *
      * @param stdClass $submission
      * @return bool
      */
     public function is_empty(stdClass $submission) {
-        return false;
+        return true;
     }
 
   /**
