diff --git a/elis/core/lib/form/autocomplete.php b/elis/core/lib/form/autocomplete.php
index e004221..2aa8297 100644
--- a/elis/core/lib/form/autocomplete.php
+++ b/elis/core/lib/form/autocomplete.php
@@ -83,7 +83,15 @@ if ($filter->_selection_enabled !== true) {
 
 if ($mode === 'search') {
 
+    echo "<script>
+          var autocelem = document.getElementById('id_{$filter->_uniqueid}');
+          if (autocelem) {
+              autocelem.value = '". ($filter->_useid ? '': $q) ."';
+          }
+          </script>\n";
+
     //search the database
+    $results = '';
     if (!empty($q)) {
         $results = $filter->get_search_results($q);
         $q = explode(' ',$q);
@@ -105,10 +113,11 @@ if ($mode === 'search') {
 
     foreach ($results as $result) {
         $add_selection_params = array(
-            'id_'.$requested_filter,
+            $requested_filter,
             $result->id,
             addslashes($filter->get_results_label($result)),
-            $filter->_ui
+            $filter->_ui,
+            $filter->_useid
         );
 
         echo '<tr onclick="add_selection(\''.implode('\',\'',$add_selection_params).'\')" class="datarow">';
@@ -119,6 +128,7 @@ if ($mode === 'search') {
         }
         echo '</tr>';
     }
+    unset($results);
 
     echo '</table>';
 
@@ -141,7 +151,6 @@ if ($mode === 'search') {
         <head>
             <script src="<?php echo $CFG->wwwroot.'/elis/core/js/jquery-1.7.1.min.js'; ?>"></script>
             <link rel="stylesheet" type="text/css" href="<?php echo $CFG->wwwroot,'/theme/styles.php?theme='.$CFG->theme?>" />
-            <link rel="stylesheet" type="text/css" href="<?php echo $CFG->wwwroot,'/theme/',$CFG->theme,'/styles.php'?>" />
             <style>
                 tr:hover{background-color:#efe;}
                 .mform {width:95%;}
