diff --git a/install/lang/vi/admin.php b/install/lang/vi/admin.php
index d07078d..c9f3c51 100644
--- a/install/lang/vi/admin.php
+++ b/install/lang/vi/admin.php
@@ -34,11 +34,5 @@ $string['clianswerno'] = 'n';
 $string['cliansweryes'] = 'y';
 $string['cliincorrectvalueerror'] = 'Lỗi, giá trị không đúng "{$a->value}" for "{$a->option}"';
 $string['cliincorrectvalueretry'] = 'Giá trị không đúng, vui lòng thử lại';
-$string['clitypevalue'] = 'nhập giá trị';
-$string['clitypevaluedefault'] = 'nhập giá trị, nhấn phím Enter để dùng giá trị mặc định ({$a})';
-$string['cliunknowoption'] = 'Lựa chọn chưa được nhận diện:
-  {$a}
-Vui lòng sử dụng lựa chọn --help.';
-$string['cliyesnoprompt'] = 'nhập y (có) hay n (không)';
 $string['environmentrequireinstall'] = 'cần phải được cài hay kích hoạt.';
 $string['environmentrequireversion'] = 'Cần phiên bản {$a->needed} trong khi bạn đang dùng {$a->current}';
