diff --git a/report/log/locallib.php b/report/log/locallib.php
index 9544e91..c69d4e8 100644
--- a/report/log/locallib.php
+++ b/report/log/locallib.php
@@ -190,12 +190,12 @@ function report_log_print_mnet_selector_form($hostid, $course, $selecteduser=0,
         $sites = array();
         if ($CFG->mnet_localhost_id == $hostid) {
             if (has_capability('report/log:view', $sitecontext) && $showcourses) {
-                if ($ccc = $DB->get_records("course", null, "fullname","id,fullname,category")) {
+                if ($ccc = $DB->get_records("course", null, "fullname","id,shortname,fullname,category")) {
                     foreach ($ccc as $cc) {
                         if ($cc->id == SITEID) {
                             $sites["$hostid/$cc->id"]   = format_string($cc->fullname).' ('.get_string('site').')';
                         } else {
-                            $courses["$hostid/$cc->id"] = format_string($cc->fullname);
+                            $courses["$hostid/$cc->id"] = format_string(get_course_display_name_for_list($cc));
                         }
                     }
                 }
@@ -281,7 +281,10 @@ function report_log_print_mnet_selector_form($hostid, $course, $selecteduser=0,
     $timemidnight = $today = usergetmidnight($timenow);
 
     // Put today up the top of the list
-    $dates = array("$timemidnight" => get_string("today").", ".userdate($timenow, $strftimedate) );
+    $dates = array(
+        "0" => get_string('alldays'),
+        "$timemidnight" => get_string("today").", ".userdate($timenow, $strftimedate)
+    );
 
     if (!$course->startdate or ($course->startdate > $timenow)) {
         $course->startdate = $course->timecreated;
@@ -295,7 +298,7 @@ function report_log_print_mnet_selector_form($hostid, $course, $selecteduser=0,
         $numdates++;
     }
 
-    if ($selecteddate == "today") {
+    if ($selecteddate === "today") {
         $selecteddate = $today;
     }
 
@@ -310,7 +313,7 @@ function report_log_print_mnet_selector_form($hostid, $course, $selecteduser=0,
         echo html_writer::select($dropdown, "host_course", $hostid.'/'.$cid);
     } else {
         $courses = array();
-        $courses[$course->id] = $course->fullname . ((empty($course->category)) ? ' ('.get_string('site').') ' : '');
+        $courses[$course->id] = get_course_display_name_for_list($course) . ((empty($course->category)) ? ' ('.get_string('site').') ' : '');
         echo html_writer::label(get_string('selectacourse'), 'menuid', false, array('class' => 'accesshide'));
         echo html_writer::select($courses,"id",$course->id, false);
         if (has_capability('report/log:view', $sitecontext)) {
@@ -355,7 +358,7 @@ function report_log_print_mnet_selector_form($hostid, $course, $selecteduser=0,
     }
 
     echo html_writer::label(get_string('date'), 'menudate', false, array('class' => 'accesshide'));
-    echo html_writer::select($dates, "date", $selecteddate, get_string("alldays"));
+    echo html_writer::select($dates, "date", $selecteddate, false);
     echo html_writer::label(get_string('showreports'), 'menumodid', false, array('class' => 'accesshide'));
     echo html_writer::select($activities, "modid", $selectedactivity, get_string("allactivities"));
     echo html_writer::label(get_string('actions'), 'menumodaction', false, array('class' => 'accesshide'));
@@ -458,10 +461,10 @@ function report_log_print_selector_form($course, $selecteduser=0, $selecteddate=
     }
 
     if (has_capability('report/log:view', $sitecontext) && $showcourses) {
-        if ($ccc = $DB->get_records("course", null, "fullname", "id,fullname,category")) {
+        if ($ccc = $DB->get_records("course", null, "fullname", "id,shortname,fullname,category")) {
             foreach ($ccc as $cc) {
                 if ($cc->category) {
-                    $courses["$cc->id"] = format_string($cc->fullname);
+                    $courses["$cc->id"] = format_string(get_course_display_name_for_list($cc));
                 } else {
                     $courses["$cc->id"] = format_string($cc->fullname) . ' (Site)';
                 }
@@ -560,7 +563,7 @@ function report_log_print_selector_form($course, $selecteduser=0, $selecteddate=
     } else {
         //        echo '<input type="hidden" name="id" value="'.$course->id.'" />';
         $courses = array();
-        $courses[$course->id] = $course->fullname . (($course->id == SITEID) ? ' ('.get_string('site').') ' : '');
+        $courses[$course->id] = get_course_display_name_for_list($course) . (($course->id == SITEID) ? ' ('.get_string('site').') ' : '');
         echo html_writer::label(get_string('selectacourse'), 'menuid', false, array('class' => 'accesshide'));
         echo html_writer::select($courses,"id",$course->id, false);
         if (has_capability('report/log:view', $sitecontext)) {
