diff --git a/elis/core/lib/data/data_filter.class.php b/elis/core/lib/data/data_filter.class.php
index 1e4d62f..02f086e 100644
--- a/elis/core/lib/data/data_filter.class.php
+++ b/elis/core/lib/data/data_filter.class.php
@@ -179,7 +179,7 @@ class OR_filter extends data_filter {
 
         $rv = array();
         if (!empty($where_clauses)) {
-            $rv['where'] = '(' . implode(")\n OR (", $where_clauses) . ')';
+            $rv['where'] = '((' . implode(")\n OR (", $where_clauses) . '))';
             $rv['where_parameters'] = $where_parameters;
         }
         return $rv;
