diff --git a/elis/core/lib/data/data_object.class.php b/elis/core/lib/data/data_object.class.php
index d8e8809..12479c2 100644
--- a/elis/core/lib/data/data_object.class.php
+++ b/elis/core/lib/data/data_object.class.php
@@ -995,6 +995,7 @@ function validate_is_unique(elis_data_object $record, array $fields) {
     $tablename = $classname::TABLE;
     $db = $record->get_db();
     $filters = array();
+
     foreach ($fields as $field) {
         $filters[] = new field_filter($field, $record->$field);
     }
