diff --git a/user/profile/field/menu/field.class.php b/user/profile/field/menu/field.class.php
index 48ddd71..21342c6 100644
--- a/user/profile/field/menu/field.class.php
+++ b/user/profile/field/menu/field.class.php
@@ -21,7 +21,8 @@ class profile_field_menu extends profile_field_base {
         }
         foreach($options as $key => $option) {
             //$this->options[$key] = format_string($option);//multilang formatting
-            $this->options[$option] = format_string($option);//multilang formatting
+            $option = format_string($option); // multilang formatting
+            $this->options[$option] = $option;
             // BJB110906: ELIS-3099, MDL-16764
         }
 
