diff --git a/mod/quiz/accessrule/password/tests/rule_test.php b/mod/quiz/accessrule/password/tests/rule_test.php
index eac66b7..5fede6f 100644
--- a/mod/quiz/accessrule/password/tests/rule_test.php
+++ b/mod/quiz/accessrule/password/tests/rule_test.php
@@ -53,6 +53,7 @@ class quizaccess_password_testcase extends basic_testcase {
             get_string('requirepasswordmessage', 'quizaccess_password'));
         $this->assertFalse($rule->prevent_new_attempt(0, $attempt));
         $this->assertFalse($rule->is_finished(0, $attempt));
-        $this->assertFalse($rule->time_left($attempt, 1));
+        $this->assertFalse($rule->end_time($attempt));
+        $this->assertFalse($rule->time_left_display($attempt, 0));
     }
 }
