diff --git a/lib/ajax/section_classes.js b/lib/ajax/section_classes.js
index ab127f3..adcf5b5 100644
--- a/lib/ajax/section_classes.js
+++ b/lib/ajax/section_classes.js
@@ -592,8 +592,10 @@ resource_class.prototype.init_resource = function(id, group, config, parentObj)
     this.id = this.getEl().id.replace(/module-/i, '');
 
     this.hidden = false;
-    if (YAHOO.util.Dom.hasClass(this.getEl().getElementsByTagName('a')[0], 'dimmed') ||
-        YAHOO.util.Dom.hasClass(this.getEl().getElementsByTagName('div')[1], 'dimmed_text')) {
+    // If dimmed and is not conditional dimmed then hidden.
+    if ((YAHOO.util.Dom.hasClass(this.getEl().getElementsByTagName('a')[0], 'dimmed') ||
+        YAHOO.util.Dom.hasClass(this.getEl().getElementsByTagName('div')[1], 'dimmed_text')) &&
+        !YAHOO.util.Dom.hasClass(this.getEl().getElementsByTagName('a')[0], 'conditionalhidden')) {
         this.hidden = true;
     }
     this.hiddenStored = null;
