diff --git a/elis/core/plugins/user_activity/db/upgrade.php b/elis/core/plugins/user_activity/db/upgrade.php
index 576920c..e7d4818 100644
--- a/elis/core/plugins/user_activity/db/upgrade.php
+++ b/elis/core/plugins/user_activity/db/upgrade.php
@@ -83,6 +83,13 @@ function xmldb_eliscoreplugins_user_activity_upgrade($oldversion) {
         upgrade_plugin_savepoint($result, 2012021400, 'eliscoreplugins','user_activity');
     }
 
+    if ($result && $oldversion < 2012081000) {
+        $sql = "UPDATE {etl_user_module_activity} SET duration = 3600 WHERE duration > 3600";
+        $DB->execute($sql);
+
+        upgrade_plugin_savepoint($result, 2012081000, 'eliscoreplugins','user_activity');
+    }
+
     return $result;
 }
 
