diff --git a/lib/phpunit/classes/hint_resultprinter.php b/lib/phpunit/classes/hint_resultprinter.php
index 33ec793..4f68206 100644
--- a/lib/phpunit/classes/hint_resultprinter.php
+++ b/lib/phpunit/classes/hint_resultprinter.php
@@ -88,11 +88,29 @@ class Hint_ResultPrinter extends PHPUnit_TextUI_ResultPrinter {
             $file = substr($file, strlen($cwd)+1);
         }
 
+        $executable = null;
+
+        if (isset($_SERVER['argv'][0])) {
+            if (preg_match('/phpunit(\.bat|\.cmd)?$/', $_SERVER['argv'][0])) {
+                $executable = $_SERVER['argv'][0];
+                for($i=1;$i<count($_SERVER['argv']);$i++) {
+                    if (!isset($_SERVER['argv'][$i])) {
+                        break;
+                    }
+                    if (in_array($_SERVER['argv'][$i], array('--colors', '--verbose', '-v', '--debug', '--strict'))) {
+                        $executable .= ' '.$_SERVER['argv'][$i];
+                    }
+                }
+            }
+        }
+
+        if (!$executable) {
         $executable = 'phpunit';
         if (phpunit_bootstrap_is_cygwin()) {
             $file = str_replace('\\', '/', $file);
             $executable = 'phpunit.bat';
         }
+        }
 
         $this->write("\nTo re-run:\n $executable $testName $file\n");
     }
