diff --git a/elis/core/plugins/user_activity/health.php b/elis/core/plugins/user_activity/health.php
index 9a200f7..894cf98 100644
--- a/elis/core/plugins/user_activity/health.php
+++ b/elis/core/plugins/user_activity/health.php
@@ -36,9 +36,16 @@ class user_activity_health_empty extends crlm_health_check_base {
     }
 
     function exists() {
-        // health warning if the cron hasn't been run, or hasn't run for the
-        // last week
-        return !$this->lastrun;
+        global $DB;
+        // health warning if the cron hasn't been run, or is more than a week behind
+        if (!$this->lastrun) {
+            return true;
+        }
+        require_once(dirname(__FILE__) .'/etl.php');
+        $state = user_activity_task_init(false);
+        $last_time = (int)$state['starttime'];
+        return($DB->count_records_select('log', "time >= $last_time") > 0 &&
+               (time() - (7 * DAYSECS)) > $last_time);
     }
 
     function title() {
