diff --git a/elis/core/plugins/user_activity/version.php b/elis/core/plugins/user_activity/version.php
index 93c139f..296e954 100644
--- a/elis/core/plugins/user_activity/version.php
+++ b/elis/core/plugins/user_activity/version.php
@@ -3,4 +3,4 @@
 defined('MOODLE_INTERNAL') || die();
 
 $plugin->version = 2012081001;
-$plugin->release = '2.2.0.19 (Build: 20130711)';
+$plugin->release = '2.2.0.19 (Build: 20130620)';
