diff --git a/install/lang/uk/admin.php b/install/lang/uk/admin.php
index 0d0180b..28de033 100644
--- a/install/lang/uk/admin.php
+++ b/install/lang/uk/admin.php
@@ -30,13 +30,5 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['clianswerno'] = 'ні';
-$string['cliansweryes'] = 'так';
-$string['cliincorrectvalueerror'] = 'Помилка, некоректне значення "{$a->value}" для "{$a->option}"';
-$string['cliincorrectvalueretry'] = 'Некоректне значення, повторіть правильно';
-$string['clitypevalue'] = 'тип значення';
-$string['clitypevaluedefault'] = 'тип значення, натисніть Enter для використання типового значення ({$a})';
-$string['cliunknowoption'] = 'Невизначені опції: {$a}. Будь ласка, використайте опцію --help.';
-$string['cliyesnoprompt'] = 'натисніть y (означає так) або n (означає ні)';
 $string['environmentrequireinstall'] = 'повинен бути встановлений і включений';
 $string['environmentrequireversion'] = 'рекомендується версія {$a->needed}, використовується версія {$a->current}';
