diff --git a/install/lang/uk/admin.php b/install/lang/uk/admin.php
index 28de033..bf18c4f 100644
--- a/install/lang/uk/admin.php
+++ b/install/lang/uk/admin.php
@@ -31,4 +31,4 @@
 defined('MOODLE_INTERNAL') || die();
 
 $string['environmentrequireinstall'] = 'повинен бути встановлений і включений';
-$string['environmentrequireversion'] = 'рекомендується версія {$a->needed}, використовується версія {$a->current}';
+$string['environmentrequireversion'] = 'потрібна версія {$a->needed} (дійсна версія {$a->current)}';
