diff --git a/question/format/blackboard_six/formatpool.php b/question/format/blackboard_six/formatpool.php
index a6c942c..7911305 100644
--- a/question/format/blackboard_six/formatpool.php
+++ b/question/format/blackboard_six/formatpool.php
@@ -88,12 +88,16 @@ class qformat_blackboard_six_pool extends qformat_blackboard_six_base {
                 array('#', 'BODY', 0, '#', 'TEXT', 0, '#'),
                 '', true, get_string('importnotext', 'qformat_blackboard_six'));
 
-        $question->questiontext = $this->cleaned_text_field($text);
-        $question->questiontextformat = FORMAT_HTML; // Needed because add_blank_combined_feedback uses it.
+        $questiontext = $this->cleaned_text_field($text);
+        $question->questiontext = $questiontext['text'];
+        $question->questiontextformat = $questiontext['format']; // Needed because add_blank_combined_feedback uses it.
+        if (isset($questiontext['itemid'])) {
+            $question->questiontextitemid = $questiontext['itemid'];
+        }
 
         // Put name in question object. We must ensure it is not empty and it is less than 250 chars.
         $id = $this->getpath($questiondata, array('@', 'id'), '',  true);
-        $question->name = $this->create_default_question_name($question->questiontext['text'],
+        $question->name = $this->create_default_question_name($question->questiontext,
                 get_string('defaultname', 'qformat_blackboard_six' , $id));
 
         $question->generalfeedback = '';
@@ -453,7 +457,7 @@ class qformat_blackboard_six_pool extends qformat_blackboard_six_base {
                 $subanswercount++;
             }
             if ($subquestioncount < 2 || $subanswercount < 3) {
-                    $this->error(get_string('notenoughtsubans', 'qformat_blackboard_six', $question->questiontext['text']));
+                    $this->error(get_string('notenoughtsubans', 'qformat_blackboard_six', $question->questiontext));
             } else {
                 $questions[] = $question;
             }
