diff --git a/question/format/blackboard_six/formatbase.php b/question/format/blackboard_six/formatbase.php
index da08e91..4424f5f 100644
--- a/question/format/blackboard_six/formatbase.php
+++ b/question/format/blackboard_six/formatbase.php
@@ -148,16 +148,4 @@ class qformat_blackboard_six_base extends qformat_based_on_xml {
     public function cleaned_text_field($text) {
         return $this->text_field($this->cleaninput($text));
     }
-
-    /**
-     * Convert the question text to plain text.
-     * We need to overwrite this function because questiontext is an array.
-     */
-    protected function format_question_text($question) {
-        global $DB;
-        $formatoptions = new stdClass();
-        $formatoptions->noclean = true;
-        return html_to_text(format_text($question->questiontext['text'],
-                $question->questiontext['format'], $formatoptions), 0, false);
-    }
 }
