diff --git a/mod/feedback/lib.php b/mod/feedback/lib.php
index 84c3273..3e1adaa 100644
--- a/mod/feedback/lib.php
+++ b/mod/feedback/lib.php
@@ -414,11 +414,6 @@ function feedback_get_recent_mod_activity(&$activities, &$index,
     }
 
     $cm_context = get_context_instance(CONTEXT_MODULE, $cm->id);
-
-    if (!has_capability('mod/feedback:view', $cm_context)) {
-        return;
-    }
-
     $accessallgroups = has_capability('moodle/site:accessallgroups', $cm_context);
     $viewfullnames   = has_capability('moodle/site:viewfullnames', $cm_context);
     $groupmode       = groups_get_activity_groupmode($cm, $course);
