diff --git a/mod/feedback/lib.php b/mod/feedback/lib.php
index 5b9c6d0..3e1adaa 100644
--- a/mod/feedback/lib.php
+++ b/mod/feedback/lib.php
@@ -397,7 +397,7 @@ function feedback_get_recent_mod_activity(&$activities, &$index,
 
     $sql .= " WHERE fc.timemodified > ? AND fk.id = ? ";
     $sqlargs[] = $timemodified;
-    $sqlargs[] = $cm->instace;
+    $sqlargs[] = $cm->instance;
 
     if ($userid) {
         $sql .= " AND u.id = ? ";
