diff --git a/mod/assign/gradingoptionsform.php b/mod/assign/gradingoptionsform.php
index b24f4fa..8eb7ce3 100644
--- a/mod/assign/gradingoptionsform.php
+++ b/mod/assign/gradingoptionsform.php
@@ -59,7 +59,7 @@ class mod_assign_grading_options_form extends moodleform {
 
         // quickgrading
         if ($instance['showquickgrading']) {
-            $mform->addElement('checkbox', 'quickgrading', get_string('quickgrading', 'assign'), '');
+            $mform->addElement('checkbox', 'quickgrading', get_string('quickgrading', 'assign'), '', $dirtyclass);
             $mform->addHelpButton('quickgrading', 'quickgrading', 'assign');
             $mform->setDefault('quickgrading', $instance['quickgrading']);
         }
