diff --git a/mod/assign/module.js b/mod/assign/module.js
index a1cbe07..ec5e071 100644
--- a/mod/assign/module.js
+++ b/mod/assign/module.js
@@ -117,9 +117,11 @@ M.mod_assign.init_grading_options = function(Y) {
             Y.one('form.gradingoptionsform').submit();
         });
         var filterelement = Y.one('#id_filter');
+        if (filterelement) {
         filterelement.on('change', function(e) {
             Y.one('form.gradingoptionsform').submit();
         });
+        }
         var quickgradingelement = Y.one('#id_quickgrading');
         if (quickgradingelement) {
             quickgradingelement.on('change', function(e) {
