diff --git a/mod/assign/renderer.php b/mod/assign/renderer.php
index 993db48..107eae8 100644
--- a/mod/assign/renderer.php
+++ b/mod/assign/renderer.php
@@ -452,7 +452,11 @@ class mod_assign_renderer extends plugin_renderer_base {
             $t->data[] = $row;
 
             foreach ($status->submissionplugins as $plugin) {
-                if ($plugin->is_enabled() && $plugin->is_visible() && !$plugin->is_empty($status->submission)) {
+                $pluginshowsummary = !$plugin->is_empty($status->submission) || !$plugin->allow_submissions();
+                if ($plugin->is_enabled() &&
+                    $plugin->is_visible() &&
+                    $pluginshowsummary) {
+
                     $row = new html_table_row();
                     $cell1 = new html_table_cell($plugin->get_name());
                     $pluginsubmission = new assign_submission_plugin_submission($plugin, $status->submission, assign_submission_plugin_submission::SUMMARY, $status->coursemoduleid, $status->returnaction, $status->returnparams);
