diff --git a/auth/mnet/auth.php b/auth/mnet/auth.php
index 0a136e8..14a42ac 100644
--- a/auth/mnet/auth.php
+++ b/auth/mnet/auth.php
@@ -378,7 +378,9 @@ class auth_plugin_mnet extends auth_plugin_base {
                 $extra = $DB->get_records_sql($sql);
 
                 $keys = array_keys($courses);
-                $defaultrole = reset(get_archetype_roles('student'));
+                $studentroles = get_archetype_roles('student');
+                if (!empty($studentroles)) {
+                    $defaultrole = reset($studentroles);
                 //$defaultrole = get_default_course_role($ccache[$shortname]); //TODO: rewrite this completely, there is no default course role any more!!!
                 foreach ($keys AS $id) {
                     if ($courses[$id]->visible == 0) {
@@ -397,6 +399,9 @@ class auth_plugin_mnet extends auth_plugin_base {
                     $courses[$id] = (array)$courses[$id];
                 }
             } else {
+                    throw new moodle_exception('unknownrole', 'error', '', 'student');
+                }
+            } else {
                 // if the array is empty, send it anyway
                 // we may be clearing out stale entries
                 $courses = array();
