diff --git a/install/lang/zh_tw/admin.php b/install/lang/zh_tw/admin.php
index ef41512..43f6dd0 100644
--- a/install/lang/zh_tw/admin.php
+++ b/install/lang/zh_tw/admin.php
@@ -30,5 +30,14 @@
 
 defined('MOODLE_INTERNAL') || die();
 
+$string['clianswerno'] = 'n';
+$string['cliansweryes'] = 'y';
+$string['cliincorrectvalueerror'] = '錯誤，不正確的值 "{$a->value}" 用於 "{$a->option}"';
+$string['cliincorrectvalueretry'] = '不正確的值，請再試一次';
+$string['clitypevalue'] = '輸入值';
+$string['clitypevaluedefault'] = '輸入值，按Enter可使用預設值({$a})';
+$string['cliunknowoption'] = '不認得的選項：  {$a}
+請使用 --幫助 選項。';
+$string['cliyesnoprompt'] = '輸入y(是) 或n(否)';
 $string['environmentrequireinstall'] = '必須安裝/啟用';
 $string['environmentrequireversion'] = '要求版本為 {$a->needed} ，您目前版本為 {$a->current}';
