diff --git a/mod/quiz/accessrule/securewindow/tests/rule_test.php b/mod/quiz/accessrule/securewindow/tests/rule_test.php
index 3bfe25d..4ee3c74 100644
--- a/mod/quiz/accessrule/securewindow/tests/rule_test.php
+++ b/mod/quiz/accessrule/securewindow/tests/rule_test.php
@@ -54,6 +54,7 @@ class quizaccess_securewindow_testcase extends basic_testcase {
         $this->assertEmpty($rule->description());
         $this->assertFalse($rule->prevent_new_attempt(0, $attempt));
         $this->assertFalse($rule->is_finished(0, $attempt));
-        $this->assertFalse($rule->time_left($attempt, 1));
+        $this->assertFalse($rule->end_time($attempt));
+        $this->assertFalse($rule->time_left_display($attempt, 0));
     }
 }
