diff --git a/elis/core/scripts/preupgrade.php b/elis/core/scripts/preupgrade.php
index ec5468d..518b714 100644
--- a/elis/core/scripts/preupgrade.php
+++ b/elis/core/scripts/preupgrade.php
@@ -251,6 +251,7 @@ if ($status) {
 mtrace(' ... '.get_string('done', 'elis_core')."!\n");
 
 
+
 /*
  * Remove two ELIS auth plugins that are no longer present if they are enabled (Alfresco SSO and ELIS dummy plugin).
  */
