diff --git a/mod/quiz/accessrule/accessrulebase.php b/mod/quiz/accessrule/accessrulebase.php
index a3d9fa1..4b585cd 100644
--- a/mod/quiz/accessrule/accessrulebase.php
+++ b/mod/quiz/accessrule/accessrulebase.php
@@ -180,15 +180,29 @@ abstract class quiz_access_rule_base {
 
     /**
      * If, because of this rule, the user has to finish their attempt by a certain time,
-     * you should override this method to return the amount of time left in seconds.
+     * you should override this method to return the attempt end time.
+     * @param object $attempt the current attempt
+     * @return mixed the attempt close time, or false if there is no close time.
+     */
+    public function end_time($attempt) {
+        return false;
+    }
+
+    /**
+     * If the user should be shown a different amount of time than $timenow - $this->end_time(), then
+     * override this method.  This is useful if the time remaining is large enough to be omitted.
      * @param object $attempt the current attempt
      * @param int $timenow the time now. We don't use $this->timenow, so we can
      * give the user a more accurate indication of how much time is left.
-     * @return mixed false if there is no deadline, of the time left in seconds if there is one.
+     * @return mixed the time left in seconds (can be negative) or false if there is no limit.
      */
-    public function time_left($attempt, $timenow) {
+    public function time_left_display($attempt, $timenow) {
+        $endtime = $this->end_time($attempt);
+        if ($endtime === false) {
         return false;
     }
+        return $endtime - $timenow;
+    }
 
     /**
      * @return boolean whether this rule requires that the attemp (and review)
