diff --git a/elis/core/fields/manual/sources/learning_objectives.php b/elis/core/fields/manual/sources/learning_objectives.php
index 1425018..f23147e 100644
--- a/elis/core/fields/manual/sources/learning_objectives.php
+++ b/elis/core/fields/manual/sources/learning_objectives.php
@@ -45,13 +45,13 @@ class manual_options_learning_objectives extends manual_options_base_class {
         } else {
             // just get ALL completion elements (LOs)
             global $DB;
-            $compelems = $DB->get_records('crlm_course_completion', null, '', 'id, name, idnumber');
+            $compelems = $DB->get_recordset('crlm_course_completion', null, '', 'id, name, idnumber');
         }
-        $compelems = $compelems ? $compelems : array();
         $result = array('' => get_string('anyvalue', 'filters'));
         foreach ($compelems as $compelem) {
             $result[$compelem->idnumber] = "{$compelem->name} ({$compelem->idnumber})";
         }
+        unset($compelems);
         return $result;
     }
 }
