diff --git a/message/lib.php b/message/lib.php
index d6ae449..b18dd61 100644
--- a/message/lib.php
+++ b/message/lib.php
@@ -1999,7 +1999,7 @@ function message_format_message($message, $format='', $keywords='', $class='othe
         $messagetext = highlight($keywords, $messagetext);
     }
 
-    return '<div class="message '.$class.'"><a name="m'.$message->id.'"></a> <span class="time">'.$time.'</span>: <span class="content">'.$messagetext.'</span></div>';
+    return '<div class="message '.$class.'"><a name="m'.$message->id.'"></a> <span class="time">'.$time.'</span>: <span class="content">'. nl2br($messagetext) .'</span></div>';
 }
 
 /**
