diff --git a/lib/blocklib.php b/lib/blocklib.php
index 9413201..841a000 100644
--- a/lib/blocklib.php
+++ b/lib/blocklib.php
@@ -1834,6 +1834,7 @@ function block_add_block_ui($page, $output) {
 
     $actionurl = new moodle_url($page->url, array('sesskey'=>sesskey()));
     $select = new single_select($actionurl, 'bui_addblock', $menu, null, array(''=>get_string('adddots')), 'add_block');
+    $select->set_label(get_string('addblock'), array('class'=>'accesshide'));
     $bc->content = $OUTPUT->render($select);
     return $bc;
 }
