diff --git a/lib/blocklib.php b/lib/blocklib.php
index 0dad6f6..ceb63af 100644
--- a/lib/blocklib.php
+++ b/lib/blocklib.php
@@ -451,7 +451,9 @@ class block_manager {
      * @return bool True if all of the blocks within that region are docked
      */
     public function region_completely_docked($region, $output) {
-        if (!$this->page->theme->enable_dock) {
+        global $CFG;
+        // If theme doesn't allow docking or allowblockstodock is not set, then return.
+        if (!$this->page->theme->enable_dock || empty($CFG->allowblockstodock)) {
             return false;
         }
 
