diff --git a/lib/moodlelib.php b/lib/moodlelib.php
index d892bb7..0cadcde 100644
--- a/lib/moodlelib.php
+++ b/lib/moodlelib.php
@@ -2984,6 +2984,9 @@ function create_user_record($username, $password, $auth='manual') {
 
     if (insert_record('user', $newuser)) {
         $user = get_complete_user_data('username', $newuser->username, $CFG->mnet_localhost_id);
+
+        set_custom_profile_fields($newuser, $authplugin, $user->id);
+
         if(!empty($CFG->{'auth_'.$newuser->auth.'_forcechangepassword'})){
             set_user_preference('auth_forcepasswordchange', 1, $user->id);
         }
@@ -2993,6 +2996,54 @@ function create_user_record($username, $password, $auth='manual') {
     return false;
 }
 
+function set_custom_profile_fields($newuser, $authplugin, $userid) {
+    if(is_object($newuser)) {
+        $newuser = get_object_vars($newuser);
+    }
+
+    $fields = array_intersect_key($newuser, array_flip($authplugin->custom_fields));
+    $fields = array_diff_key($fields, array_flip($authplugin->userfields));
+
+    foreach($fields as $key=>$value) {
+        $info_field = get_record('user_info_field', 'shortname', $key);
+
+        $data = get_field('user_info_data', 'data', 'userid', $userid, 'fieldid', $info_field->id);
+
+        if($data === false) {
+            $data = $info_field->defaultdata;
+        }
+
+        if(strcmp($data, $value) !== 0) {
+            $valid = true;
+
+                //check to make sure that the value we are placing in is a valid one
+            if(strcmp($info_field->datatype, 'menu') == 0){
+                $validValues = explode("\n", $info_field->param1);
+                if(!in_array($value, $validValues)) {
+                    $valid = false;
+                }
+            } else if(strcmp($info_field->datatype, 'checkbox') == 0) {
+                if($value != 1 && $value != 0) {
+                    $valid = false;
+                }
+            }
+
+            if($valid) {
+                if(record_exists('user_info_data', 'userid', $userid, 'fieldid', $info_field->id)) {
+                    set_field('user_info_data', 'data', $value, 'userid', $userid, 'fieldid', $info_field->id);
+                } else {
+                    $data = new object();
+                    $data->userid = $userid;
+                    $data->fieldid = $info_field->id;
+                    $data->data = $value;
+
+                    insert_record('user_info_data', $data);
+                }
+            }
+        }
+    }
+}
+
 /**
  * Will update a local user record from an external source
  *
@@ -3226,6 +3277,7 @@ function authenticate_user_login($username, $password) {
 
             if (!$authplugin->is_internal()) {            // update user record from external DB
                 $user = update_user_record($username, get_auth_plugin($user->auth));
+                events_trigger('user_updated', $user);
             }
         } else {
             // if user not found, create him
