diff --git a/lib/questionlib.php b/lib/questionlib.php
index 311629d..1cf1620 100644
--- a/lib/questionlib.php
+++ b/lib/questionlib.php
@@ -1803,10 +1803,10 @@ function question_pluginfile($course, $context, $component, $filearea, $args, $f
         send_file_not_found();
     }
 
+    if ($filearea === 'export') {
     list($context, $course, $cm) = get_context_info_array($context->id);
     require_login($course, false, $cm);
 
-    if ($filearea === 'export') {
         require_once($CFG->dirroot . '/question/editlib.php');
         $contexts = new question_edit_contexts($context);
         // check export capability
