diff --git a/lib/setup.php b/lib/setup.php
index ecb2050..8a8b255 100644
--- a/lib/setup.php
+++ b/lib/setup.php
@@ -230,8 +230,8 @@ if (!defined('AJAX_SCRIPT')) {
 // Begin Remote-Learner edit
 // detect NanoGong plugin
 if (!defined('NANOGONG')) {
-    if (file_exists($CFG->libdir.'/editor/tinymce/tiny_mce/3.5.7b/plugins/nanogong/nanogong.php')) {
-        define('NANOGONG', '3.5.7b');
+    if (file_exists($CFG->libdir.'/editor/tinymce/tiny_mce/3.5.1.1/plugins/nanogong/nanogong.php')) {
+        define('NANOGONG', '3.5.1.1');
     }
 }
 // End Remote-Learner edit
