diff --git a/lib/tablelib.php b/lib/tablelib.php
index 859b055..31fdee1 100644
--- a/lib/tablelib.php
+++ b/lib/tablelib.php
@@ -734,16 +734,19 @@ class flexible_table {
     function col_fullname($row) {
         global $COURSE, $CFG;
 
-        if (!$this->download) {
-            $profileurl = new moodle_url('/user/profile.php', array('id' => $row->{$this->useridfield}));
-            if ($COURSE->id != SITEID) {
-                $profileurl->param('course', $COURSE->id);
+        $name = fullname($row);
+        if ($this->download) {
+            return $name;
             }
-            return html_writer::link($profileurl, fullname($row));
 
+        $userid = $row->{$this->useridfield};
+        if ($COURSE->id == SITEID) {
+            $profileurl = new moodle_url('/user/profile.php', array('id' => $userid));
         } else {
-            return fullname($row);
+            $profileurl = new moodle_url('/user/view.php',
+                    array('id' => $userid, 'course' => $COURSE->id));
         }
+        return html_writer::link($profileurl, $name);
     }
 
     /**
