diff --git a/elis/core/fields/manual/field_controls/menu.php b/elis/core/fields/manual/field_controls/menu.php
index 319f474..0cd96d5 100644
--- a/elis/core/fields/manual/field_controls/menu.php
+++ b/elis/core/fields/manual/field_controls/menu.php
@@ -26,8 +26,9 @@ function menu_control_display($form, $mform, $customdata, $field, $as_filter=fal
                 $options = array('' => get_string("choose"));
             }
             foreach($tmpoptions as $key => $option) {
-                $option = trim($option);
-                $options[$option] = format_string($option);//multilang formatting
+                $option = trim($option, "\n\r");
+                $option = format_string($option);
+                $options[$option] = $option;//multilang formatting
             }
         } else {
             $source = $manual->param_options_source;
