diff --git a/admin/auth_config.php b/admin/auth_config.php
index 0281b12..4203db4 100644
--- a/admin/auth_config.php
+++ b/admin/auth_config.php
@@ -51,6 +51,9 @@ if ($frm = data_submitted() and confirm_sesskey()) {
 }
 
 $user_fields = $authplugin->userfields;
+//BEGIN1 RL Canada edit
+$custom_fields = $authplugin->custom_fields;
+//END1 RL Canada edit
 //$user_fields = array("firstname", "lastname", "email", "phone1", "phone2", "institution", "department", "address", "city", "country", "description", "idnumber", "lang");
 
 /// Get the auth title (from core or own auth lang files)
@@ -70,11 +73,38 @@ echo "<input type=\"hidden\" name=\"auth\" value=\"".$auth."\" />\n";
 // auth plugin description
 print_simple_box_start('center', '80%');
 print_heading($authtitle);
+
+// ELIS release version
+if (!property_exists($authplugin, 'release')) {
+    // check for version.php file in auth directory
+    $versionfile = "{$CFG->dirroot}/auth/{$auth}/version.php";
+    if (file_exists($versionfile)) {
+        if (is_readable($versionfile)) {
+            $module = new stdClass;
+            $plugin = new stdClass;
+            include($versionfile);
+            if (property_exists($module, 'release')) {
+                $authplugin->release = $module->release;
+            } else if (property_exists($plugin, 'release')) {
+                $authplugin->release = $plugin->release;
+            }
+        } else {
+            error_log("admin/auth_config.php: version file '$versionfile' not readable!");
+        }
+    }
+}
+if (property_exists($authplugin, 'release')) {
+    echo '<center><div class="settingsform elisrelease"><p>';
+    echo get_string('elis_release', 'block_elis_info').$authplugin->release;
+    echo '</p></div></center>';
+}
+
 print_simple_box_start('center', '80%', '', 5, 'informationbox');
 echo $authdescription;
 print_simple_box_end();
 echo "<hr />\n";
-$authplugin->config_form($frm, $err, $user_fields);
+//EDIT2 RL Canada edit added($custom_fields)
+$authplugin->config_form($frm, $err, $user_fields, $custom_fields);
 print_simple_box_end();
 echo '<p style="text-align: center"><input type="submit" value="' . get_string("savechanges") . "\" /></p>\n";
 echo "</div>\n";
@@ -89,7 +119,8 @@ exit;
 // but some may want a custom one if they are offering
 // other options
 // Note: lockconfig_ fields have special handling.
-function print_auth_lock_options ($auth, $user_fields, $helptext, $retrieveopts, $updateopts) {
+//EDIT3 RL Canada added $custom_fields = array()
+function print_auth_lock_options ($auth, $user_fields, $helptext, $retrieveopts, $updateopts, $custom_fields = array()) {
 
     echo '<tr><td colspan="3">';
     if ($retrieveopts) {
@@ -176,6 +207,65 @@ function print_auth_lock_options ($auth, $user_fields, $helptext, $retrieveopts,
         }
         echo '</tr>';
     }
+
+    //BEGIN4 RL Canada edit
+    if(!empty($custom_fields)) {
+        echo '<tr><td colspan="3">';
+
+        print_heading('Custom Fields'); //language string
+
+        echo '</td></tr>';
+
+        foreach($custom_fields as $field) {
+
+            // Define some vars we'll work with
+            if (!isset($pluginconfig->{"field_map_$field"})) {
+                $pluginconfig->{"field_map_$field"} = '';
+            }
+            if (!isset($pluginconfig->{"field_updatelocal_$field"})) {
+                $pluginconfig->{"field_updatelocal_$field"} = '';
+            }
+            if (!isset($pluginconfig->{"field_updateremote_$field"})) {
+                $pluginconfig->{"field_updateremote_$field"} = '';
+            }
+            if (!isset($pluginconfig->{"field_lock_$field"})) {
+                $pluginconfig->{"field_lock_$field"} = '';
+            }
+
+            // define the fieldname we display to the  user
+            $fieldname = get_field('user_info_field', 'name', 'shortname', $field);
+
+            if ($retrieveopts) {
+                $varname = 'field_map_' . $field;
+
+                echo '<tr valign="top"><td align="right">';
+                echo '<label for="lockconfig_'.$varname.'">'.$fieldname.'</label>';
+                echo '</td><td>';
+
+                echo "<input id=\"lockconfig_{$varname}\" name=\"lockconfig_{$varname}\" type=\"text\" size=\"30\" value=\"{$pluginconfig->$varname}\" />";
+                echo '<div style="text-align: right">';
+                echo '<label for="menulockconfig_field_updatelocal_'.$field.'">'.get_string('auth_updatelocal', 'auth') . '</label>&nbsp;';
+                choose_from_menu($updatelocaloptions, "lockconfig_field_updatelocal_{$field}", $pluginconfig->{"field_updatelocal_$field"}, "");
+                echo '<br />';
+                if ($updateopts) {
+                    echo '<label for="menulockconfig_field_updateremote_'.$field.'">'.get_string('auth_updateremote', 'auth') . '</label>&nbsp;';
+                    choose_from_menu($updateextoptions, "lockconfig_field_updateremote_{$field}", $pluginconfig->{"field_updateremote_$field"}, "");
+                    echo '<br />';
+                }
+                echo '<label for="menulockconfig_field_lock_'.$field.'">'.get_string('auth_fieldlock', 'auth') . '</label>&nbsp;';
+                choose_from_menu($lockoptions, "lockconfig_field_lock_{$field}", $pluginconfig->{"field_lock_$field"}, "");
+                echo '</div>';
+            } else {
+                echo '<tr valign="top"><td align="right">';
+                echo '<label for="menulockconfig_field_lock_'.$field.'">'.$fieldname.'</label>';
+                echo '</td><td>';
+                choose_from_menu($lockoptions, "lockconfig_field_lock_{$field}", $pluginconfig->{"field_lock_$field"}, "");
+            }
+            echo '</td>';
+            echo '</tr>';
+        }
+    }
+    //END4 RL-Canada edit
 }
 
 ?>
