diff --git a/admin/renderer.php b/admin/renderer.php
index d99092d..d2117c4 100644
--- a/admin/renderer.php
+++ b/admin/renderer.php
@@ -374,7 +374,7 @@ class core_admin_renderer extends plugin_renderer_base {
         $copyrighttext = '<a href="http://moodle.org/">Moodle</a> '.
                          '<a href="http://docs.moodle.org/dev/Releases" title="'.$CFG->version.'">'.$CFG->release.'</a><br />'.
                          'Copyright &copy; 1999 onwards, Martin Dougiamas<br />'.
-                         'and <a href="http://docs.moodle.org/dev/Credits">many other contributors</a>.<br />'.
+                         'and <a href="http://moodle.org/dev">many other contributors</a>.<br />'.
                          '<a href="http://docs.moodle.org/dev/License">GNU Public License</a>';
         //////////////////////////////////////////////////////////////////////////////////////////////////
 
@@ -708,11 +708,22 @@ class core_admin_renderer extends plugin_renderer_base {
 
                 // ELIS_2.0.0 Release
                 $versionfile = $plugin->rootdir .'/version.php';
+                $module = new stdClass; // ELIS-7928: was getting previous value
                 if (file_exists($versionfile)) {
                     include($versionfile);
                 }
-                $release = new html_table_cell(!empty($plugin->release)
-                                               ? $plugin->release : ''); // TBD
+
+                // ELIS-7214 -- Activity modules use a $module object in their version.php file so check that for a release
+                //              property as well.
+                if (isset($plugin->release)) {
+                    $rel = $plugin->release;
+                } else if (isset($module->release)) {
+                    $rel = $module->release;
+                } else {
+                    $rel = '';
+                }
+
+                $release = new html_table_cell($rel);
 
                 $isenabled = $plugin->is_enabled();
                 if (is_null($isenabled)) {
