diff --git a/install/lang/cy/admin.php b/install/lang/cy/admin.php
index 675d91c..a9fde51 100644
--- a/install/lang/cy/admin.php
+++ b/install/lang/cy/admin.php
@@ -30,15 +30,5 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['clianswerno'] = 'n';
-$string['cliansweryes'] = 'i';
-$string['cliincorrectvalueerror'] = 'Gwall, gwerth anghywir "{$a->value}" ar gyfer "{$a->option}"';
-$string['cliincorrectvalueretry'] = 'Gwerth anghywir, rhowch gynnig arall arni';
-$string['clitypevalue'] = 'math o werth';
-$string['clitypevaluedefault'] = 'math o werth, pwyswch Enter i ddefnyddio\'r gwerth diofyn ({$a})';
-$string['cliunknowoption'] = 'Dewisiadau dieithr:
-  {$a}
-Defnyddiwch y dewis --help.';
-$string['cliyesnoprompt'] = 'mae teipio i (yn golygu iawn) neu n (yn golygu na)';
 $string['environmentrequireinstall'] = 'angen ei osod/ei alluogi.';
 $string['environmentrequireversion'] = 'Mae angen fersiwn {$a->needed} ac rydych yn defnyddio {$a->current}';
