diff --git a/elis/core/db/upgrade.php b/elis/core/db/upgrade.php
index 0ff0db0..1f0366b 100644
--- a/elis/core/db/upgrade.php
+++ b/elis/core/db/upgrade.php
@@ -359,9 +359,13 @@ function xmldb_elis_core_upgrade($oldversion=0) {
             }
 
             // remove any old instances
+            if ($dbman->table_exists(new XMLDBTable('block_instance_old'))) {
             $DB->delete_records('block_instance_old', array('blockid' => $eiblock->id));
+            }
             // remove any old pinned blocks
+            if ($dbman->table_exists(new XMLDBTable('block_pinned_old'))) {
             $DB->delete_records('block_pinned_old', array('blockid' => $eiblock->id));
+            }
             // remove block record
             $DB->delete_records('block', array('id' => $eiblock->id));
 
