diff --git a/install/lang/ca_valencia/error.php b/install/lang/ca_valencia/error.php
index 3301110..9df0f04 100644
--- a/install/lang/ca_valencia/error.php
+++ b/install/lang/ca_valencia/error.php
@@ -30,5 +30,5 @@
 
 defined('MOODLE_INTERNAL') || die();
 
-$string['cannotsavemd5file'] = 'No s\'ha pogut guardar el fitxer md5';
-$string['cannotsavezipfile'] = 'No s\'ha pogut guardar el fitxer zip';
+$string['cannotsavemd5file'] = 'No s\'ha pogut alçar el fitxer md5';
+$string['cannotsavezipfile'] = 'No s\'ha pogut alçar el fitxer zip';
