diff --git a/mod/wiki/lib.php b/mod/wiki/lib.php
index 290cb4e..56dc18d 100644
--- a/mod/wiki/lib.php
+++ b/mod/wiki/lib.php
@@ -536,7 +536,7 @@ function wiki_extend_navigation(navigation_node $navref, $course, $module, $cm)
             $node = $navref->add(get_string('view', 'wiki'), $link, navigation_node::TYPE_SETTING);
         }
 
-        if (has_capability('mod/wiki:editpage', $context)) {
+        if (wiki_user_can_edit($subwiki)) {
             $link = new moodle_url('/mod/wiki/edit.php', array('pageid' => $pageid));
             $node = $navref->add(get_string('edit', 'wiki'), $link, navigation_node::TYPE_SETTING);
         }
