diff --git a/mod/lesson/essay_form.php b/mod/lesson/essay_form.php
index 650f326..ce2df3b 100644
--- a/mod/lesson/essay_form.php
+++ b/mod/lesson/essay_form.php
@@ -53,6 +53,7 @@ class essay_grading_form extends moodleform {
         $mform->addElement('hidden', 'mode', 'update');
         $mform->setType('mode', PARAM_ALPHA);
 
+        $mform->addElement('static', 'question', get_string('question', 'lesson'));
         $mform->addElement('static', 'studentanswer', get_string('studentresponse', 'lesson', fullname($this->_customdata['user'], true)));
 
         $mform->addElement('textarea', 'response', get_string('comments', 'lesson'), array('rows'=>'15', 'cols'=>'60'));
