diff --git a/enrol/manual/externallib.php b/enrol/manual/externallib.php
index f2daf62..1804a19 100644
--- a/enrol/manual/externallib.php
+++ b/enrol/manual/externallib.php
@@ -90,7 +90,7 @@ class enrol_manual_external extends external_api {
 
             //throw an exception if user is not able to assign the role
             $roles = get_assignable_roles($context);
-            if (!key_exists($enrolment['roleid'], $roles)) {
+            if (!array_key_exists($enrolment['roleid'], $roles)) {
                 $errorparams = new stdClass();
                 $errorparams->roleid = $enrolment['roleid'];
                 $errorparams->courseid = $enrolment['courseid'];
